service/collector/error.go

```diff
@@ -1,10 +1,20 @@
 package collector
 
 import (
+	"strings"
+
 	"github.com/aws/aws-sdk-go/aws/awserr"
 	"github.com/giantswarm/microerror"
 )
 
+// IsEndpointNotAvailable asserts that an error is due to service
+// not available in the current region.
+func IsEndpointNotAvailable(err error) bool {
+	c := microerror.Cause(err)
+
+	return strings.Contains(c.Error(), "no such host")
+}
```

Comment:
FYI this caused a panic on `gauss`. We had to fix this for our release now. See https://github.com/giantswarm/aws-operator/pull/2462.

Reply:
```
$ k -n giantswarm logs --tail 1000 -f aws-operator-8-6-0-79ffc9bcbc-t92zn | gg
{
    "caller": "github.com/giantswarm/aws-operator/service/collector/helper.go:148",
    "level": "debug",
    "message": "collecting metrics in account: 107744613923",
    "time": "09:01:40"
}
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0x1b3b38f]

goroutine 1610 [running]:
github.com/giantswarm/aws-operator/service/collector.IsEndpointNotAvailable(0x0, 0x0, 0xc000b24c48)
	/root/project/service/collector/error.go:15 +0x3f
github.com/giantswarm/aws-operator/service/collector.getDefaultVPCQuotaFor(0x22a5307, 0xa, 0xc0000a61f0, 0xc0000a6200, 0x27e58e0, 0xc0000a6220, 0x27ca560, 0xc0000a6318, 0x27d7580, 0xc0000a6400, ...)
	/root/project/service/collector/servicequota.go:156 +0x96
github.com/giantswarm/aws-operator/service/collector.(*ServiceQuota).collectForAccount(0xc0004ec9f0, 0xc000f00120, 0xc0000a61f0, 0xc0000a6200, 0x27e58e0, 0xc0000a6220, 0x27ca560, 0xc0000a6318, 0x27d7580, 0xc0000a6400, ...)
	/root/project/service/collector/servicequota.go:131 +0x167
github.com/giantswarm/aws-operator/service/collector.(*ServiceQuota).Collect.func1(0x0, 0x139c65a)
	/root/project/service/collector/servicequota.go:97 +0xa3
golang.org/x/sync/errgroup.(*Group).Go.func1(0xc001324720, 0xc001322580)
	/go/pkg/mod/golang.org/x/sync@v0.0.0-20190911185100-cd5d95a43a6e/errgroup/errgroup.go:57 +0x59
created by golang.org/x/sync/errgroup.(*Group).Go
	/go/pkg/mod/golang.org/x/sync@v0.0.0-20190911185100-cd5d95a43a6e/errgroup/errgroup.go:54 +0x66
```

Reply:
These types of issues are frustrating for all involved. But I get frustrated because they are easily detectable with an integration test.

We could use the `basicapp` test here. It tests that the helm chart can be installed and the deployment is ready which would have caught this.

https://github.com/giantswarm/e2etests/tree/54179310a7c62c3ab0baa9304c014352574098a6/basicapp

It runs in a KIND cluster in under 5 mins. I agree testing cluster creation is better in a control plane but we should still have integration tests IMO.

Reply:
Hey ross this is a good shout. I will check this out and see what I can come up with. MY stance is that I do not want to write full metal integration tests with cluster creation and that shebang anymore, but instead go for custom conformance tests on a real Control Plane. But this here would make sense to have I guess.

Reply:
sorry for the hassle, for some reason I missed it during my tests :(

Reply:
Thanks Tim let me know if I can help.

I'd love it if we replaced the current tests in integration with the basicapp test.

We have it on our list in https://github.com/giantswarm/giantswarm/issues/11035 to update helmclient to 1.0.0. That will also get rid of all the replace crap from go.mod. 🎉